### DISCUSSION TITLE
Can I create long short portfolio using any of Hierarchical Risk Parity (HRP) algorithms

### DISCUSSION THREAD

**Discussion by thomsonian2023** [dcajasn/Riskfolio-Lib#81] [Q&A]
I am reading through the examples and it is not immediately clear how to use any of the HRP class algorithms to create a long-short portfolio.

Thanks

**Comment by dcajasn** [comment]
Hi @thomsonian2023,

By the moment it is not posible. The way PortfolioLab library makes that is defining in advance which assets take short positions. For me the problem with this, is that the algo doesn't calculate short positions internally.

Best,
Dany

**Reply by thomsonian2023** [reply]
Do you know of any other similar python packages implementing clustering but calculates short positions also?

**Reply by dcajasn** [reply]
The only one is PortfolioLab, but it's not available on pypi, you must look for PortfolioLab 0.3.0 mirror on google. On the other hand, I don't know papers about short positions on hierarchical risk parity.

**Reply by BradKML** [reply]
Wanted to look around and seek (FOSS) solutions for these two "longshort" experiments:
- https://policytensor.substack.com/p/youre-probably-underweight-real-estate
- https://policytensor.substack.com/p/how-to-systematically-beat-the-60
